src/core/system_parameters/osnma_data.h

```diff
@@ -121,31 +119,26 @@ class MACK_message
     MACK_header header;
     std::vector<MACK_tag_and_info> tag_and_info;
     std::vector<uint8_t> key;
-    uint32_t TOW; // TODO duplicated variable, also in NavData
+    uint32_t TOW;  // TODO duplicated variable, also in OSNMA_NavData
     uint32_t WN;
     uint32_t PRNa;
 };
 
-class NavData
+class OSNMA_NavData
 {
 public:
-    NavData()=default;
-    void init(const std::shared_ptr<OSNMA_msg> &osnma_msg);
-    std::vector<uint8_t> ephemeris_iono_vector{};
-    std::string ephemeris_iono_vector_2{};
-    std::vector<uint8_t> utc_vector{};
-    std::string utc_vector_2{};
-    uint32_t PRNa{};
-    uint32_t WN_sf0{};
-    uint32_t TOW_sf0{};
-private:
-    Galileo_Ephemeris EphemerisData;
-    Galileo_Iono IonoData;
-    Galileo_Utc_Model UtcData;
-    void generate_eph_iono_vector(); // TODO pass data directly fro Telemetry Decoder (if bits are in the needed order)
-    void generate_utc_vector(); // TODO
-
+    OSNMA_NavData() = default;
+    void init(const std::shared_ptr<OSNMA_msg>& osnma_msg);
+    std::string get_ephemeris_iono_data() const;
+    std::string get_utc_data() const;
+    uint32_t get_tow_sf0() const;
```

Comment:
Carles, should all the public members in the osnma_data file be replaced with getters and setters?

Reply:
Well, it is not strictly needed, but it has some minor advantages:

- The get method are marked const, so they can be safely used in other methods that are also const.
- You can also add extra constraints/checks to the set methods (not done here)
- If we need to expand that class in the future, we can do it without breaking things.

None of those reasons are highly relevant here, to be honest, but the price we pay is negligible.